contrib/babelfishpg_common/src/numeric.c

```diff
@@ -1029,3 +1064,253 @@ tsql_numeric_get_typmod(Numeric num)
 
 	return (((precision & 0xFFFF) << 16 ) | (scale & 0xFFFF)) + VARHDRSZ;
 }
+
+Datum
+bigint_poly_sum(PG_FUNCTION_ARGS)
+{
+	
+	PolyNumAggState *state;
+	
+	state = PG_ARGISNULL(0) ? NULL : (PolyNumAggState *) PG_GETARG_POINTER(0);
+	/* If there were no non-null inputs, return NULL */
+	if (state == NULL || state->N == 0)
+		PG_RETURN_NULL();
+
+	#ifdef HAVE_INT128
+	int128 			arg;
+	arg = state->sumX;
+
+	if (unlikely(arg < PG_INT64_MIN) || unlikely(arg > PG_INT64_MAX))
+		ereport(ERROR,
+				(errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
+				 errmsg("Arithmetic overflow error converting expression to data type bigint.")));
+
+	PG_RETURN_INT64((int64) arg);
+	#else
+
+	Datum temp;
+	bool flag;
+	int64 result;
+	NumericVar nvar;
+
+	temp = DirectFunctionCall1(numeric_sum, PG_GETARG_DATUM(0));
+	init_var(&nvar);
+	set_var_from_num( DatumGetNumeric(temp), &nvar);
```

Comment:
We don't need this one.. Just copy and extend the functionality of numeric_sum to return NumericVar

Reply:
Changed in the latest commit